feat(repo/cache): add s3 support for user configured folder hierarchy (#18865)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

    AWS_REGION
```

> <!-- prettier-ignore -->
> !!! tip
> When using S3 for repository cache storage, a folder hierarchy can be used by specifying it as part of `repositoryCacheType`.
> e.g. `repositoryCacheType: 's3://bucket-name/dir1/.../dirN/'`.

Reviewer: ```suggestion
<!-- prettier-ignore -->
!!! tip
    If you're storing the repository cache on Amazon S3 then you may set a folder hierarchy as part of `repositoryCacheType`.
    For example, `repositoryCacheType: 's3://bucket-name/dir1/.../dirN/'`.
```

The admonition needs four spaces. 😉 And we can use simpler words.
